app/forms/link_case/confirmation_form.rb

```diff
@@ -0,0 +1,16 @@
+module LinkCase
+  class ConfirmationForm < BaseForm
+    form_for LinkedApplication
+
+    attr_accessor :link_type_code
+
+    validates :link_type_code, presence: true, unless: proc { draft? }
+
```

Comment:
As discussed we could potentially destroy the link here using 

```ruby
after_validation do
  linked_application.destroy! unless link_type_code
end
```

Reply:
If I destroy the model after validation, won't it fail when it gets to the `model.update` line in the `save` method?